test_responses.py

```diff
@@ -54,6 +56,34 @@ def run():
     assert_reset()
 
 
+def test_response_with_custom_reason():
+    @responses.activate
+    def run():
+        responses.add(
+            responses.GET,
+            "http://example.com",
+            body=b"take care!",
+            status=299,
+            reason="warning",
+        )
+        requests.get("http://example.com")
```

Comment:
I think we need to just test the newly added feature. something like:

```python
        resp = requests.get("http://example.com")
        assert resp.status_code == 299
        assert resp.reason == "warning"

```

Reply:
We can only test the newly added features. In my code I checked all values just to make an entire test. I think it's at least not worse to test the other values :)